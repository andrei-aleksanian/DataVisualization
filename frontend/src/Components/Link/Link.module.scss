.index {
  border: var(--color-action) solid 3px;
  border-radius: 0.5rem;
  background-color: var(--color-action);
  color: var(--color-text);
  transition: 0.3s all ease-in-out;
  cursor: pointer;
  text-decoration: none;
}

.Action {
  display: block;
  text-align: center;
  font-size: 1.6rem;
  padding: 0.5em 1em;
}

.Back {
  border-radius: 0 0 0.5rem 0;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 60px;
  height: 35px;
  position: absolute;
  top: 0;
  left: 0;
  z-index: var(--z-index-10);
}

.Back img {
  height: 15px;
}

.Block {
  position: relative;
}

.external {
  text-decoration: none;
}
