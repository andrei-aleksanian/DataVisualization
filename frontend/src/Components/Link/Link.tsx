import { Link } from 'react-router-dom';
import cx from 'classnames';
import arrow from '../assets/arrow.svg';
import classes from './Link.module.scss';

interface LinkProps {
  link: string;
  customClass?: string;
}

interface LinkBackProps extends LinkProps {
  block?: boolean;
}

interface LinkHeroProps extends LinkProps {
  text: string;
  external?: boolean;
}

export const defaultPropsLink = {
  customClass: '',
  external: false,
};
/*
Big Hero Link. Used for call to Action. Rediercts the user to a page of your choice.
*/
export const LinkHero = ({ text, link, external, customClass }: LinkHeroProps) => {
  return (
    <>
      {external ? (
        <a
          className={cx(classes.index, classes.Action, classes.external, customClass)}
          href={link}
          target="_blank"
          rel="noreferrer"
        >
          {text}
        </a>
      ) : (
        <Link to={link} className={cx(classes.index, classes.Action, customClass)}>
          {text}
        </Link>
      )}
    </>
  );
};
LinkHero.defaultProps = defaultPropsLink;

export const defaultPropsLinkBack = {
  customClass: '',
  block: false,
};
/*
Small link back with an arrow. Used to come back to a page of your choice.
Intended use is to go back but can accept any link.
*/
export const LinkBack = ({ link, customClass, block }: LinkBackProps) => (
  <Link to={link} className={cx(classes.index, classes.Back, { [classes.Block]: block }, customClass)}>
    <img src={arrow} alt="back" />
  </Link>
);
LinkBack.defaultProps = defaultPropsLinkBack;
