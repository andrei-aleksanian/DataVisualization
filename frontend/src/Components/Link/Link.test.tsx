import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { LinkBack, LinkHero } from './Link';

test('LinkBack matches snapshot', async () => {
  const { asFragment } = render(<LinkBack link="/" />, { wrapper: MemoryRouter });
  expect(asFragment()).toMatchSnapshot();
});

test('LinkHero matches snapshot', async () => {
  const { asFragment } = render(<LinkHero link="/" text="test link" />, { wrapper: MemoryRouter });
  expect(asFragment()).toMatchSnapshot();
});
