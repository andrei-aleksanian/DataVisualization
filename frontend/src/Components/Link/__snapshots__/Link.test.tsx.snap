// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkBack matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="index Back"
    href="/"
  >
    <img
      alt="back"
      src="arrow.svg"
    />
  </a>
</DocumentFragment>
`;

exports[`LinkHero matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="index Action"
    href="/"
  >
    test link
  </a>
</DocumentFragment>
`;
